app/api/models.py

```diff
@@ -0,0 +1,25 @@
+from pydantic import BaseModel
+
+
+# NOTE: Because we have multiple successful response models, we need extra="forbid" to
+# ensure that instances of SuccessfulUploadWithWarning are not unintentionally
+# converted/filtered to the parent SuccessfulUpload by FastAPI when returning a response.
```

Comment:
Hm, I don't think so. In the CLI, the data dictionary models already use this parameter, e.g. https://github.com/neurobagel/bagel-cli/blob/1fb6380448fd6d70579e8ab314674351a66ba46e/bagel/dictionary_models.py#L37-L38.

IIRC the extra `"identifies"` field (if that's what you're referring to) was because we needed a way to differentiate columns with ID labels from 'normal' continuous/categorical columns. I think that problem remains even if you forbid extra keys.